org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java

```diff
@@ -1541,14 +1542,15 @@ public void fire(IJavaElementDelta customDelta, int eventType) {
 			listenerMask = this.state.elementChangedListenerMasks;
 			listenerCount = this.state.elementChangedListenerCount;
 		}
-
-		switch (eventType) {
-			case DEFAULT_CHANGE_EVENT:
-			case ElementChangedEvent.POST_CHANGE:
-				firePostChangeDelta(deltaToNotify, listeners, listenerMask, listenerCount);
-				fireReconcileDelta(listeners, listenerMask, listenerCount);
-				break;
-		}
+		Job.create("Fire Change", m->{ //$NON-NLS-1$
+			switch (eventType) {
+				case DEFAULT_CHANGE_EVENT:
+				case ElementChangedEvent.POST_CHANGE:
+					firePostChangeDelta(deltaToNotify, listeners, listenerMask, listenerCount);
+					fireReconcileDelta(listeners, listenerMask, listenerCount);
+					break;
+			}
+		}).schedule();
```

Comment:
This is a "no go". There will be N jobs running in arbitrary order after this change.

Reply:
Given that `flush` is called on the `this` instance from within `firePostChangeDelta` I feel that this will break all sorts of invariants. Random async changes to a code base that is not designed for asynchronous processing is not trivial.